validator_client/src/beacon_node_fallback.rs

```diff
@@ -499,7 +547,7 @@ impl<T: SlotClock, E: EthSpec> BeaconNodeFallback<T, E> {
 
                 // There exists a race condition where `func` may be called when the candidate is
                 // actually not ready. We deem this an acceptable inefficiency.
-                match func(&$candidate.beacon_node).await {
+                match func($candidate.beacon_node.clone()).await {
```

Comment:
Any reason why we need to clone the `BeaconNodeHttpClient` here instead of passing it a reference? Perhaps we want to `Arc` the `beacon_node` field if we really need ownership (each clone is 296 bytes)?